Sources/Datadog/URLSessionAutoInstrumentation/Interception/URLSessionInterceptor.swift

```diff
@@ -95,10 +93,12 @@ public class URLSessionInterceptor: URLSessionInterceptorType {
         guard !internalURLsFilter.isInternal(url: request.url) else {
             return request
         }
-        let ddDelegate = session?.delegate as? DDURLSessionDelegate
-        let firstPartyURLsFilter = ddDelegate?.firstPartyURLsFilter ?? defaultFirstPartyURLsFilter
-        if injectTracingHeadersToFirstPartyRequests,
-           firstPartyURLsFilter.isFirstParty(url: request.url) {
+        let delegateURLFilter = (session?.delegate as? DDURLSessionDelegate)?.firstPartyURLsFilter
+        let isFirstPartyForDelegate = (delegateURLFilter?.isFirstParty(url: request.url)) ?? false
+        let isFirstPartyForInterceptor = self.defaultFirstPartyURLsFilter.isFirstParty(url: request.url)
+        let isFirstParty = isFirstPartyForDelegate || isFirstPartyForInterceptor
```

Comment:
How about decoupling it to single `isFirstPartyRequest(request: Request, delegate: URLSessionDelegate?) -> Bool`? Now we repeat the same logic in two places.